### PR TITLE
[Do not merge] Added Release Lease to Data Lake Flush

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.DataLake/swagger/readme.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ## Configuration
 ``` yaml
 # Generate file storage
-input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/storage-dataplane-preview/specification/storage/data-plane/Microsoft.StorageDataLake/stable/2020-02-10/DataLakeStorage.json
+input-file: C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.StorageDataLake\stable\2020-06-12\DataLakeStorage.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/swagger/readme.md:7]: I will revert this before merging.

File: sdk/storage/generate.ps1 [hunk 2]
```diff
@@ -2,17 +2,17 @@ pushd $PSScriptRoot/Azure.Storage.Common/swagger/Generator/
 npm install
 npm install -g autorest@beta
 
-cd $PSScriptRoot/Azure.Storage.Blobs/swagger/
-autorest --use=$PSScriptRoot/Azure.Storage.Common/swagger/Generator/ --verbose
+#cd $PSScriptRoot/Azure.Storage.Blobs/swagger/
```
Comment (seanmcc-msft) [inline sdk/storage/generate.ps1:5]: I will revert this before merging.

**Review by seanmcc-msft — COMMENTED** [review]
I didn't add `ReleaseLease` to `DataLakeOpenWriteOptions` or `DataLakeFileUploadOptions`

- We call Flush internally whenever the user calls `DataLakeOpenWriteStream.Flush()`.  I don't think it makes sense to be sending `ReleaseLease` each time the user calls Flush.
- For `DataLakeFileClient.Upload()`, we don't have an overload that takes both `overwrite` and `DataLakeRequestConditions`.  This means it won't be possible to pass in the `LeaseId`, which is required if you specify `ReleaseLease = true`.

**Review by kasobol-msft — APPROVED** [review]

**Comment by jsquire** [comment]
Hi @seanmcc-msft.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream by removing the `no-recent-activity` label?  Otherwise, we'll close this out in 7 days.

**Comment by seanmcc-msft** [comment]
Please leave this PR open.

**Pull request closed without merging**